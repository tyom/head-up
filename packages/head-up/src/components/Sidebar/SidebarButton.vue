<template>
  <button
    v-bind="$attrs"
    type="button"
    class="button"
    v-on="$listeners"
  >
    <slot/>
  </button>
</template>

<style scoped>
.button {
  width: 100%;
  padding: 0.7em;
  background: none;
  text-align: center;
  line-height: 1;
  color: inherit;
  border: 0;
  outline: 0;

  &:hover {
    color: #fff;
    background-color: rgba(#fff, 0.05);
  }

  &.focus-visible {
    box-shadow: 0 0 0 1px #fffa;
    background-color: rgba(#fff, 0.05);
  }

  &:active {
    color: #fff;
    animation: bounce-1 0.1s forwards;
    background-color: #2b5691;
  }

  &._active {
    background-color: #20416d;
    color: #fff;
    animation: none;

    &:active {
      background-color: #12253e;
    }
  }
}
</style>
