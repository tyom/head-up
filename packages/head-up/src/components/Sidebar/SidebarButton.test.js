import { shallowMount } from '@vue/test-utils';
import SidebarButton from './SidebarButton';

test('render default', () => {
  const wrapper = shallowMount(SidebarButton, {
    slots: {
      default: 'Clicky',
    },
  });
  expect(wrapper).toMatchSnapshot();
});
