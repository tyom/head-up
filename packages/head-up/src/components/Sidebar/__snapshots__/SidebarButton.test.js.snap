// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render default 1`] = `<button type="button" class="button">Clicky</button>`;
